### ISSUE TITLE
sjsxc prevents SOGo V3 normal login process display. Breaks things on wrong password.

### ISSUE THREAD

**Issue by popindavibe** [jsxc/jsxc#456]
The problem is still here with last versions of both SOGo V3 and SJSXC. 
To be honest it never really integrated seamlessly regarding the authentication process. But it got worse with latest versions of SOGo, and now if a user enters a wrong password, it just turns green and stays that way, although when deactivating JSXC SOGo turns orange in case of error and ask if you want to try again, and display 'success' when you successfully log in.

So now that it does not show the form back, it is really starting to create a problem and confusing users.

**Comment by popindavibe** [comment]
Some more info. I tried updating JSXC via git clone instead of zip release on another SOGo server. The links are the same for v2 and v3, and eventually althouth I took link from v3 tag 3.1.1, there was no jquery.min.js and css folder was in /build althouth looked for in /
I think it takes v2 basically, and it broke the JSXC plugin after the repository cloning (nothing displays although no error in console).

**Comment by sualko** [comment]
I can reproduce your first issue with SOGo 3.2.6, but currently I have no clue why this is happening.

If you clone the repo make sure to checkout the [sogo-v3](https://github.com/jsxc/jsxc.sogo/tree/sogo-v3) branch.

**Comment by popindavibe** [comment]
Ok thanks, I'm still getting acquainted with the whole git/ github stuff.

For SOGo v3, now that you confirmed, I'll try to look for clues, I can just guess that there must be some JS mix happening on the validation step, between Sogo / angularJS and JSXC. 
Will get back with info (if I find any).

**Comment by popindavibe** [comment]
Alright, don't know how it all fit together, but if I forbid access to `jsxc.sogo.css`, then I get expected behaviour at login (but nothing is showing for JSXC once I'm in, which I guess is also expected behaviour in that case).

**Comment by sualko** [comment]
> So CSS issue...?

No. JSXC waits for the stylesheet and is "disabled" until it is loaded, but I have already an idea why it is not working. So stay tuned.

**Comment by sualko** [comment]
I was able to [fix the login](https://github.com/jsxc/jsxc.sogo/commit/b03d111205dd602059573d214241d39b97b5522b) (kind of), but the animation is still broken. I think I have to talk to the inverse guys to get a clean solution.

**Comment by popindavibe** [comment]
Thanks, I'll try the fix and hope you'll be able to sort things out with the SOGo team.

**Comment by popindavibe** [comment]
Well, the fix is indeed working, although I confirm animation is kinda broken and it's not so easy to understand whether you managed to log in or not (you end up waiting extra seconds each time).
A seamless integration would be fantastic. Did you manage to get in touch with Inverse?

**Comment by sualko** [comment]
> Did you manage to get in touch with Inverse?

Yes I am in contact with those guys. We will see what happens :wink: 

> there is no dev folder anymore

This folder will only appear after you built your own dev version.

Anyway thank you very much for testing.

**Comment by popindavibe** [comment]
Hi Klaus,
Bringing bad news!

On Firefox 45.6 I'm getting JS errors : 
![pb_sjsxc](https://cloud.githubusercontent.com/assets/7958775/23262938/6de74c8e-f9d4-11e6-8a36-bc73d8023a9f.png)

But it tests fine on Firefox 49 Android. I think it is the patch that broke compatibility, but not 100% sure.

**Comment by popindavibe** [comment]
Well, Firefox 51 got the same result on Windows (js errors and no JSXC loading). Now I'm very confused!
Working fine on Chrome, pointing back at **mozRTCPeerConnection** as a pure Firefox compatibility issue.

**Comment by popindavibe** [comment]
well, trying back from laptop it works but some links were missing related to the build folder.
@sualko when you say "This folder will only appear after you built your own dev version." Is there an easy to do that and get everything at the right place straight from `git clone` ?

I've added extra symbolic links (for sound and other SVG files), will try again tomorrow at work to see if it makes a change. I use the same version of Firefox (45.6) though, so this is all the more strange. I'm starting to wonder if something is not getting tampered with at work...

**Comment by sualko** [comment]
>  Is there an easy to do that

Go to `js/jsxc`, execute `make` and copy the build folder to your desired location.

**Comment by popindavibe** [comment]
Hmm, my ̀js/jsxc` folder contained nothing at all:
```
./
|-- jsxc
|   |-- img -> ../../build/js/jsxc/img
|   |-- lib -> ../../build/js/jsxc/lib
|   `-- sound -> ../../build/js/jsxc/sound
|-- lib
|   |-- jquery.min.js
|   `-- jquery.ui.min.js
|-- old
|   |-- sjsxc.js.bak
|   `-- sjsxc.js.orig
|-- sjsxc.config.js
|-- sjsxc.config.sample.js
`-- sjsxc.js
```
From the top of sjsxc folder:

```
./                                                                                                                     
|-- CHANGELOG.md                                                                                                       
|-- Gruntfile.js                                                                                                       
|-- LICENSE                                                                                                            
|-- README.md                                                                                                          
|-- ajax                                                                                                               
|   |-- cgi-bin                                                                                                        
|   `-- getturncredentials.php                                                                                         
|-- build                                                                                                              
|   |-- LICENSE                                                                                                        
|   |-- ajax                                                                    
|   |-- css
|   |-- img
|   `-- js (empty, where I created the symlinks)
|-- css
|   |-- jquery-ui.min.css
|   `-- jsxc.sogo.css
|-- img
|   |-- chat-icon-white.svg
|   |-- close.svg
|   |-- closelabel.png
|   |-- edit.png
|   |-- gear.png
|   |-- gear_hover.png
|   |-- loading.gif
|   |-- padlock_close_green.png
|   |-- padlock_close_grey.png
|   |-- padlock_close_orange.png
|   `-- padlock_open.png
|-- js
|   |-- jsxc
|   |-- lib
|   |-- old
|   |-- sjsxc.config.js
|   |-- sjsxc.config.sample.js
|   `-- sjsxc.js
|-- package.json
|-- scss
|   |-- _colors.scss
|   |-- _sogo.scss
|   `-- jsxc.sogo.scss
```
I'm guessing there should be a `MAKEFILE` where I have nothing.

Here below is some git info:
git logs : 
```
633c148 Merge branch 'patch-sogo-v3-1' of https://github.com/jsxc/jsxc.sogo
4ba390a Merge remote-tracking branch 'refs/remotes/origin/sogo-v3'                                          
7cbdaa2 test
b03d111 fix login function
7b041b9 build v3.1.1
a157afc update change log
ebb2a56 update jsxc
c958e03 build v3.1.0
...
```

```
git remote  show origin
* remote origin
  Fetch URL: https://github.com/jsxc/jsxc.sogo.git
  Push  URL: https://github.com/jsxc/jsxc.sogo.git
  HEAD branch: master
  Remote branches:
    master          tracked
    patch-sogo-v3-1 tracked
    sogo-v3         tracked
  Local branch configured for 'git pull':
    master merges with remote patch-sogo-v3-1
  Local ref configured for 'git push':
    master pushes to master (local out of date)
```
I reproduced it by testing locally, and still nothing in `js/jsxc`.

**Comment by popindavibe** [comment]
Just a quick note : the WebRTC error about **mozRTCPeerConnection** is linked to [issue 471](https://github.com/jsxc/jsxc/issues/471). It seems like some defaults have been changed, cause I'm positive I did not go by myself to modify the setting in about:config.
But anyway so that's  fixed.

Now only login integration and the building from cloning remain. (Sorry for the rather lengthy thread).

**Comment by sualko** [comment]
You have to execute `git submodule update --init --recursive` in your sjsxc folder in order to checkout jsxc. Afterwards you can create a fresh build of jsxc if you execute `make` inside the jsxc folder. Now you have two possibilities:

1) Change your [sogo config](https://github.com/jsxc/jsxc/wiki/Install-sjsxc-(SOGo)) in order to use the nightly version of jsxc
2) Build your own jsxc and sogo version (with `grunt build:release` in both projects) and link the sjsxc/build folder to your WebServerResources directory.

**Comment by popindavibe** [comment]
all right. Sorry for my noobyness, appreciate the help.

**Comment by puuu** [comment]
Are there any news about this issue? Did the SOGo team could help out?

I changed to SoGo v3 and want to activate jsxc again, but I am a little afraid of this issue and the fact that there was no update of jsxc.sogo to v3.2.x.

**Comment by sualko** [comment]
> Did the SOGo team could help out?

They tried, but I forgot to send them a minimal example so they can reproduce the issue. In the current version the login works fine, but the animation has some issues. I try to release a beta version today, so you can test it and if I have time I will work on the snippet for the sogo guys.

**Comment by sualko** [comment]
@puuu please test https://github.com/jsxc/jsxc.sogo/releases/tag/s3-v3.2.1-beta.1

**Comment by puuu** [comment]
Thanks, the beta works almost and after login it integrates nicely in sogo v3. Good job!

My issues are:
1. As known, the login animation is broken (e.g. it plays not completely and shows again the login screen for a short time) :-(.
2. If you first try to login with a wrong password, a green animation is show, then the login screen appears again. If you then type in the correct password, a red animation with "wrong password" warning appears, but SOGo logs in and shows e.g. the mail list, but jsxc is not connected in.

**Comment by popindavibe** [comment]
Hi, after some time out, I should get some time to do some further Sogo/ sjsxc testing. I did not manage to compile from repo (multiple npm error on a Debian Jessie), but I downloaded and installed the latest version from the zip file.

As of now, v3.2.1, I can't display sjsxc, and JS debug shows nothing. It is very mysterious.
I can get it working if I use the old sjsxc.js file though (with some minor display issues, which also might have come from last SOGo update).

Here is the config difference between the two: https://pastebin.com/cEJKehh7
